### ISSUE TITLE
Parameter of forward pass (missing?) 

### ISSUE THREAD

**Issue by lukasfrank** [huggingface/transfer-learning-conv-ai#56]
I'm studying the model and was wondering about a few things:

1. the dataloader append the padding tokens and while the training there is no attention_mask set. is there any reason for that? 
2. In your [blog post](https://medium.com/huggingface/how-to-build-a-state-of-the-art-conversational-ai-with-transfer-learning-2d818ac26313) you describe that the model should care about the positions of the tokens. Is there any reason why the position_ids parameter is not set? 
3. lm_labels parameter: according to the [documentation](https://huggingface.co/transformers/model_doc/gpt2.html#transformers.GPT2DoubleHeadsModel) labels are ignored if they are set to -100. Is that a type in the documentation or did I miss something?

**Comment by sshleifer** [comment]
1. Don't know.
2. `position_ids` are inferred by `model.forward` if `position_ids=None` (the default) is passed.
3. This code also uses a slightly older version of the repo than the docs you linked. If you read the docstring for pytorch-transformers (the older version) it says "All labels set to ``-1`` are ignored (masked)".

**Comment by silverriver** [comment]
@lukasfrank 
For your first question:

All the attention in transfertransfo is masked self attention, that means all the future tokens are masked. Therefore all the padding tokens are masked when applying the future mask, and these padding tokens will not be involved in the attention calculation process.
